core/src/main/java/io/jenkins/pluginhealth/scoring/probes/HasUnreleasedProductionChangesProbe.java

```diff
@@ -66,15 +68,14 @@ public ProbeResult doApply(Plugin plugin, ProbeContext context) {
         if (!matcher.find()) {
             return ProbeResult.error(key(), "SCM link doesn't match GitHub plugin repositories");
         }
-
-        final String folder = matcher.group("folder");
+        final Optional<Path> folder = context.getScmFolderPath();
```

Comment:
I am not entirely convinced we can really store the folder path as a `Path`. Did you make sure that with this implementation the `Path` object does contain only the folder path and not some root element as well?